pkg/hedgedmetrics/metrics.go

```diff
@@ -1,27 +1,63 @@
 package hedgedmetrics
 
 import (
+	"sync"
 	"time"
 
 	"github.com/cristalhq/hedgedhttp"
 	"github.com/prometheus/client_golang/prometheus"
 )
 
 const (
-	hedgedMetricsPublishDuration = 10 * time.Second
+	PublishDuration = 10 * time.Second
 )
 
-// PublishHedgedMetrics flushes metrics from hedged requests every 10 seconds
-func Publish(s *hedgedhttp.Stats, counter prometheus.Counter) {
-	ticker := time.NewTicker(hedgedMetricsPublishDuration)
+// CounterWithValue wraps prometheus.Counter and keeps track of the current value.
+type CounterWithValue struct {
+	counter prometheus.Counter
+	value   int64
+	mu      sync.Mutex
+}
+
+func NewCounterWithValue(counter prometheus.Counter) *CounterWithValue {
+	return &CounterWithValue{counter: counter}
+}
+
+func (c *CounterWithValue) Add(v int64) {
+	c.mu.Lock()
```

Comment:
this is only accessed from one goroutine. doesn't need locking.

Reply:
removed.

Reply:
I am seeing a DATARACE in the test `TestCounterWithValue` after removing the locking,

Reply:
That's b/c in the test you are checking the value of the counter in a different goroutine than is changing the value, but this doesn't happen in the actual application.